### PR TITLE
Improve config file schema docs

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Description
Depends on #2193, part of #2166 

Needs some CSS work but already better than the raw schema IMO

https://deploy-preview-2196--eksctl.netlify.app/usage/schema/

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

**Comment by michaelbeaumont** [comment]
One thing this doesn't show yet is the type of the entry.

File: pkg/apis/eksctl.io/v1alpha5/types.go [hunk 1]
```diff
@@ -372,7 +372,7 @@ type ClusterMeta struct {
 	// +optional
 	Version string `json:"version,omitempty"`
 	// +optional
-	Tags map[string]string `json:"tags,omitempty"`
+	Tags map[string]string `json:"tags,omitempty" jsonschema:"description=EKS Cluster tags"`
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/apis/eksctl.io/v1alpha5/types.go:375]: No but I think that can be as the next step. I'll just leave out the descriptions entirely.

**Comment by cPu1** [comment]
I found a small issue in the preview but otherwise LGTM.

When you click on a field, it navigates to the permalink for that field but the clicked field is hidden behind the header.
image

File: userdocs/src/usage/schema.json [hunk 2]
```diff
@@ -0,0 +1 @@
...pe":"boolean"},"fsx":{"type":"boolean"},"efs":{"type":"boolean"},"albIngress":{"type":"boolean"},"xRay":{"type":"boolean"},"cloudWatch":{"type":"boolean"}},"additionalProperties":false,"type":"object"},"NodeGroupInstancesDistribution":{"properties":{"instanceTypes":{"items":{"type":"string"},"type":"array"},"maxPrice":{"type":"number"},"onDemandBaseCapacity":{"type":"integer"},"onDemandPercentageAboveBaseCapacity":{"type":"integer"},"spotInstancePools":{"type":"integer"},"spotAllocationStrategy":{"type":"string"}},"additionalProperties":false,"type":"object"},"NodeGroupSGs":{"required":["withShared","withLocal"],"properties":{"attachIDs":{"items":{"type":"string"},"type":"array"},"withShared":{"type":"boolean"},"withLocal":{"type":"boolean"}},"additionalProperties":false,"type":"object"},"NodeGroupSSH":{"required":["allow"],"properties":{"allow":{"type":"boolean"},"publicKeyPath":{"type":"string"},"publicKey":{"type":"string"},"publicKeyName":{"type":"string"},"sourceSecurityGroupIds":{"items":{"type":"string"},"type":"array"}},"additionalProperties":false,"type":"object"},"SecretsEncryption":{"properties":{"keyARN":{"type":"string"}},"additionalProperties":false,"type":"object"}}}
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply userdocs/src/usage/schema.json:1]: I was going to do this as part of https://github.com/weaveworks/eksctl/issues/2059 (e.g. as a subcommand of `eksctl`?)
With this PR change the schema is no longer really discoverable for users anyway
And before it wasn't correct enough to actually use.

File: pkg/apis/eksctl.io/v1alpha5/types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/apis/eksctl.io/v1alpha5/types.go:375]: Done

**Comment by michaelbeaumont** [comment]
> I found a small issue in the preview but otherwise LGTM.

Yeah... Maybe if @bia has time, she can take a look and improve this whole view after this is merged?

**Pull request merged**